### DISCUSSION TITLE
Gitlab MR not created - 404 repository not found

### DISCUSSION THREAD

**Discussion by Ketec** [WeblateOrg/weblate#12406] [Q&A]
It successfully creates the branch and pushes changes, but for the MR it shows 404:
```
celery-celery stderr | [2024-09-03 09:03:15,509: INFO/12714] Acquired Lock('lock:lock:vcs:api:gitlab:0').
celery-celery stderr | [2024-09-03 09:03:15,509: INFO/ForkPoolWorker-6] Acquired Lock('lock:lock:vcs:api:gitlab:0').
celery-celery stderr | [2024-09-03 09:03:16,481: WARNING/12714] Could not push the repo: RepositoryError: Could not create pull request 404: 404 Project Not Found (-1)
celery-celery stderr | [2024-09-03 09:03:16,481: WARNING/ForkPoolWorker-6] Could not push the repo: RepositoryError: Could not create pull request 404: 404 Project Not Found (-1)
celery-notify stderr | [2024-09-03 09:03:16,485: INFO/MainProcess] Task weblate.accounts.tasks.notify_change[065843dc-1644-4448-baaa-024c2a33b99b] received
```
Log does not show any actual url either.
Using ssh push url, no port specified + push branch.

`git@gitlab.domain.com:group1/subgroup/project.git`

Self hosted weblate and gitlab.

I cannot see anything different or anything that i missed reading the documentation.

We have gitlab v17.2.2 and it supports rest API v4.

**Comment by nijel** [comment]
It might be caused by using subgroup, Weblate might wrongly generate the API URL in that case. Can you check GitLab logs to see which URL Weblate accesses and guess how it should be corrected?

**Reply by nijel** [reply]
PS: Weblate will log the attempted URL as well in debug mode (it will be shown without that in future releases, as I've just changed the severity of that log in 9148c82cf6be94feb92d82109251b3f6835f41d4).

**Reply by nijel** [reply]
There is https://github.com/WeblateOrg/weblate/issues/12439 covering this as well, please follow up there if you have any useful findings.

**Comment by Jahetthana** [comment]
armsomboon.cb.id
